client/lxd_instances.go

```diff
@@ -442,13 +442,40 @@ func (r *ProtocolLXD) tryCreateInstance(req api.InstancesPost, urls []string, op
 	return &rop, nil
 }
 
+func (r *ProtocolLXD) isCurrentServer(server Server) (bool, error) {
```

Comment:
Can we make the function just return a `bool`, no `error` and return `false` if it gets an error from GetConnectionInfo?

Currently `GetConnectionInfo` can't return an error anyway, so this should be fine and will make it much easier to use (can go straight in the if statement).